symfony/client/src/orangehrmPimPlugin/components/SaveMembership.vue

```diff
@@ -0,0 +1,183 @@
+            <oxd-input-field
+              type="select"
+              label="Subscription Paid By"
+              v-model="membership.subscriptionPaidBy"
+              :options="paidBy"
+            />
+          </oxd-grid-item>
+          <oxd-grid-item>
+            <oxd-input-field
+              label="Subscription Amount"
+              v-model="membership.subscriptionFee"
+              :rules="rules.subscriptionFee"
+            />
+          </oxd-grid-item>
+          <oxd-grid-item>
+            <oxd-input-field
+              type="select"
+              label="Currency"
+              v-model="membership.currencyType"
+              :options="currencies"
+            />
+          </oxd-grid-item>
+          <oxd-grid-item>
+            <oxd-input-field
+              label="Subscription Commence Date"
+              v-model="membership.subscriptionCommenceDate"
+              type="date"
+              placeholder="yyyy-mm-dd"
+              :rules="rules.subscriptionCommenceDate"
+            />
+          </oxd-grid-item>
+          <oxd-grid-item>
+            <oxd-input-field
+              label="Subscription Renewal Date"
+              v-model="membership.subscriptionRenewalDate"
+              type="date"
+              :years="yearArray"
+              placeholder="yyyy-mm-dd"
+              :rules="rules.subscriptionRenewalDate"
+            />
+          </oxd-grid-item>
+        </oxd-grid>
+      </oxd-form-row>
+
+      <oxd-form-actions>
+        <required-text />
+        <oxd-button
+          type="button"
+          displayType="ghost"
+          label="Cancel"
+          @click="onCancel"
+        />
+        <submit-button />
+      </oxd-form-actions>
+    </oxd-form>
+  </div>
+  <oxd-divider />
+</template>
+
+<script>
+import {
+  digitsOnly,
+  validDateFormat,
+  endDateShouldBeAfterStartDate,
+} from '@orangehrm/core/util/validation/rules';
+import {yearRange} from '@orangehrm/core/util/helper/year-range';
+
+const membershipModel = {
+  membershipId: [],
+  subscriptionFee: '',
+  subscriptionPaidBy: null,
+  currencyTypeId: [],
+  subscriptionCommenceDate: '',
+  subscriptionRenewalDate: '',
+};
+
+export default {
+  name: 'save-membership',
+
+  emits: ['close'],
+
+  props: {
+    http: {
+      type: Object,
+      required: true,
+    },
+    currencies: {
+      type: Array,
+      default: () => [],
+    },
+    paidBy: {
+      type: Array,
+      default: () => [],
+    },
+    memberships: {
+      type: Array,
+      default: () => [],
+    },
+  },
+
+  data() {
+    return {
+      isLoading: false,
+      membership: {...membershipModel},
+      yearArray: [...yearRange()],
+      rules: {
+        subscriptionRenewalDate: [
+          validDateFormat(),
+          endDateShouldBeAfterStartDate(
+            () => this.membership.subscriptionCommenceDate,
+            'Renewal date should be after the commencing date',
+          ),
+        ],
+        subscriptionFee: [digitsOnly],
```

Comment:
add max validations `Should be less than 1000,000,000.00` [check](https://github.com/orangehrm/orangehrm/blob/5.0/symfony/client/src/orangehrmPimPlugin/components/SaveSkill.vue#L116)